Add confirmDismiss example

Taym95 opened the pull request:
## Description

I added an example to show how to use the confirmDismiss property of Dismissible.

## Related Issues

#29318 

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [ ] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

Commit:
Add confirmDismiss example

Commit:
Add confirmDismiss example

engine-flutter-autoroll committed:
Roll engine 250b99bd275a..d4d4883216fd (3 commits) (#29742)


https://github.com/flutter/engine/compare/250b99bd275a...d4d4883216fd

git log 250b99bd275ad7622f239a8042f6a6d31b5b95f1..d4d4883216fd1c1dfc17d41418a960c002cfff57 --no-merges --oneline
d4d488321 Roll src/third_party/skia 174a3f61fd5d..a695da930510 (2 commits) (flutter/engine#8242)
2d5376c09 Roll src/third_party/skia 2c4ceca62d27..174a3f61fd5d (1 commits) (flutter/engine#8239)
b05399901 Roll src/third_party/skia e409441aee0d..2c4ceca62d27 (1 commits) (flutter/engine#8238)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

Commit:
fix asset reloading (#29469)

christopherfujino committed:
Fix more tests for ANSI terminals (#29699)

* have tests that log error messages use a FakePlatform that does not support Ansi color, to fix tests started from color terminals
* add the override to the other tests too, in case someone copy & pastes them in the future for a test involving an error message

shaynekellyii committed:
Update DropdownButton underline to be customizable (#29138)

amirh committed:
Enable platform views for Flutter Gallery on iOS. (#29632)

This results in running Flutter Gallery with one less thread (platform
and gpu threads are the the same).

This is very likely to cause a regression in the iOS Gallery benchmarks.
I'm mainly interested in landing this to see how much the benchmarks
regress, and it's likely that we will revert it shortly after landing.

Commit:
Revert "Enable platform views for Flutter Gallery on iOS. (#29632)" (#29754)

Commit:
Enable code generation features in tool  (#29399)

mdebbar committed:
Avoid flickering while dragging to select text (#29563)

Commit:
Allowing adding/updating packages during hot reload (#29747)

Commit:
Update upgrade to rebase and stash local changes. (#29192)

rubensdemelo committed:
Include brackets on OutlineButton doc (#29654)

goderbauer committed:
Fix transition to stock details (#29638)

xster committed:
Add heart shapes to CupertinoIcons (#29630)

Commit:
Use Dart version in script cache check (#29721)

Commit:
update fuchsia-attach (#29764)

engine-flutter-autoroll committed:
Roll engine d4d4883216fd..b68f7e72aa71 (7 commits) (#29776)


https://github.com/flutter/engine/compare/d4d4883216fd...b68f7e72aa71

git log d4d4883216fd1c1dfc17d41418a960c002cfff57..b68f7e72aa71eb75969bd674d78a5b40ee5ed89f --no-merges --oneline
b68f7e72a Allow specifying an alternate Mac host SDK. (flutter/engine#8249)
3df694300 Roll src/third_party/dart cabaa78cc5..ba0147ac07 (3 commits) (flutter/engine#8248)
69c4d82f5 Roll src/third_party/skia d6ab77a2e20f..19fb3971bca5 (17 commits) (flutter/engine#8245)
06635d38a Mirror Android platform views a11y tree in the Flutter a11y tree. (flutter/engine#8237)
28433c513 FlutterJNI no longer asserts it is attached when dispatching platform messages and instead fizzles with a warning if not attached. Not sure what root cause of issue is, but this is necessary to avoid crashes. (flutter/engine#8246)
1addfb596 Roll src/third_party/dart 5e9df35a57..cabaa78cc5 (36 commits) (flutter/engine#8244)
13e3b7101 Roll src/third_party/skia a695da930510..d6ab77a2e20f (11 commits) (flutter/engine#8243)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

amirh committed:
Revert "Update upgrade to rebase and stash local changes. (#29192)" (#29780)

This reverts commit e38be671a77b6c09dceb74ad2e63cc4b8671c83d.

abhijeethp committed:
Removes unnecessary "new" in documentation (#29779)

knezzz committed:
Update CupertinoTextField (#29008)

dnfield committed:
Detect and cleanup leaky processes (#29196)

* Detect and cleanup leaky processes
* Add flaky tests for detecting leaked processes

a14n committed:
some formatting of map, parameters and spaces (#29760)

mdebbar committed:
Add support for text selection via mouse to Cupertino text fields (#29769)

Commit:
roll engine to 5f8ae420c1ac61bbbb26e61251d129c879fc788d (#29817)

engine-flutter-autoroll committed:
Roll engine 5f8ae420c1ac..f8e3a40f9ba7 (2 commits) (#29830)


https://github.com/flutter/engine/compare/5f8ae420c1ac...f8e3a40f9ba7

git log 5f8ae420c1ac61bbbb26e61251d129c879fc788d..f8e3a40f9ba701120724b5d6e8ff790d50f18083 --no-merges --oneline
f8e3a40f9 Roll buildroot to 9c7b023ff266ee58b00fe60326fa1db910a087f3 (flutter/engine#8277)
c12696cc5 Roll src/third_party/skia f9264bac4e99..33332f1e100c (27 commits) (flutter/engine#8275)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
62c88d690 Roll src/third_party/skia 33332f1e100c..e86cc5039026 (1 commits) (flutter/engine#8278) (#29832)


https://github.com/flutter/engine/compare/f8e3a40f9ba7...62c88d690272

git log f8e3a40f9ba701120724b5d6e8ff790d50f18083..62c88d690272f0b78047758bb9d8ffa8bbb870cc --no-merges --oneline
62c88d690 Roll src/third_party/skia 33332f1e100c..e86cc5039026 (1 commits) (flutter/engine#8278)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
9f878fcb3 Roll src/third_party/skia e86cc5039026..ec1c6fdfdd11 (1 commits) (flutter/engine#8279) (#29836)


https://github.com/flutter/engine/compare/62c88d690272...9f878fcb3ecd

git log 62c88d690272f0b78047758bb9d8ffa8bbb870cc..9f878fcb3ecdcc2c6ba0c6d958282c176f008183 --no-merges --oneline
9f878fcb3 Roll src/third_party/skia e86cc5039026..ec1c6fdfdd11 (1 commits) (flutter/engine#8279)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
Roll engine 9f878fcb3ecd..d2bba21af27a (6 commits) (#29869)


https://github.com/flutter/engine/compare/9f878fcb3ecd...d2bba21af27a

git log 9f878fcb3ecdcc2c6ba0c6d958282c176f008183..d2bba21af27a8d3f30c66aa014f324c786f3fb4e --no-merges --oneline
d2bba21af Roll src/third_party/skia ab4c138c0ef3..d79c95e8bcf2 (2 commits) (flutter/engine#8286)
cee2bd732 Roll src/third_party/skia 97d957d1db65..ab4c138c0ef3 (3 commits) (flutter/engine#8285)
ef3cbe8ac Roll src/third_party/skia 4c0b5c4b79dd..97d957d1db65 (1 commits) (flutter/engine#8284)
bd4f6d726 Roll src/third_party/skia b1dbbc8509c6..4c0b5c4b79dd (2 commits) (flutter/engine#8283)
8d23b21d1 Roll src/third_party/skia 61bd91237778..b1dbbc8509c6 (2 commits) (flutter/engine#8282)
61838cab6 Roll src/third_party/skia ec1c6fdfdd11..61bd91237778 (1 commits) (flutter/engine#8280)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
dcc72c717 Roll src/third_party/skia d79c95e8bcf2..e9a53fe3526a (1 commits) (flutter/engine#8287) (#29874)


https://github.com/flutter/engine/compare/d2bba21af27a...dcc72c7176bc

git log d2bba21af27a8d3f30c66aa014f324c786f3fb4e..dcc72c7176bcaa9fc9fa3798542f764ca6fe4ebc --no-merges --oneline
dcc72c717 Roll src/third_party/skia d79c95e8bcf2..e9a53fe3526a (1 commits) (flutter/engine#8287)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

Commit:
dont fail build if codegen fails (#29818)

Commit:
Update upgrade to reset off of hotfix branches (#29786)

Commit:
wrap Timeline calls in assert (#29861)

DanTup committed:
Update Twitter handle @flutterio -> @FlutterDev (#29908)

The @flutterio account now just directs people to @FlutterDev:

https://twitter.com/flutterio

engine-flutter-autoroll committed:
Roll engine dcc72c7176bc..3a445cec8a51 (4 commits) (#29924)


https://github.com/flutter/engine/compare/dcc72c7176bc...3a445cec8a51

git log dcc72c7176bcaa9fc9fa3798542f764ca6fe4ebc..3a445cec8a51056e5903c12659b6fc97138e57db --no-merges --oneline
3a445cec8 Roll src/third_party/skia cd54466473fe..42197c0bc68e (10 commits) (flutter/engine#8292)
fcc63e7d0 Roll src/third_party/skia 2e2fa9e3a06b..cd54466473fe (2 commits) (flutter/engine#8291)
1b1209881 Roll src/third_party/skia c9f746cfb9f4..2e2fa9e3a06b (1 commits) (flutter/engine#8290)
b5813917f Roll src/third_party/skia e9a53fe3526a..c9f746cfb9f4 (1 commits) (flutter/engine#8288)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
Roll engine 3a445cec8a51..dd6be2fd8371 (2 commits) (#29937)


https://github.com/flutter/engine/compare/3a445cec8a51...dd6be2fd8371

git log 3a445cec8a51056e5903c12659b6fc97138e57db..dd6be2fd83710de8392fb554927a5d495684a13e --no-merges --oneline
dd6be2fd8 Roll src/third_party/skia 42197c0bc68e..15dc59e65cfa (19 commits) (flutter/engine#8294)
d6ead18f4 Make it easy to write embedder unit tests by creating a fixture and config builder. (flutter/engine#8276)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

liyuqian committed:
Remove tranparent paint hack from BackdropFilter (#29929)

We can remove the hack because https://github.com/flutter/engine/pull/8183
has been rolled into the framework.

engine-flutter-autoroll committed:
Roll engine dd6be2fd8371..af64f725f6c7 (3 commits) (#29944)


https://github.com/flutter/engine/compare/dd6be2fd8371...af64f725f6c7

git log dd6be2fd83710de8392fb554927a5d495684a13e..af64f725f6c74f5d1287a8be8511691119354088 --no-merges --oneline
af64f725f Fixing links between higher and lower classes. (flutter/engine#8295)
22ee8ee08 Migrate existing embedder unit tests to use the fixture. (flutter/engine#8296)
345ae7d37 Delegate a11y events and action to/from embedded Android platform views. (flutter/engine#8250)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

Commit:
Fix cache location, artifacts, and re-enable dart2js test (#29783)

engine-flutter-autoroll committed:
Roll engine af64f725f6c7..0d83a2ecd1bb (2 commits) (#29948)


https://github.com/flutter/engine/compare/af64f725f6c7...0d83a2ecd1bb

git log af64f725f6c74f5d1287a8be8511691119354088..0d83a2ecd1bb78f8e1f0ed9719ff3751cdfdd064 --no-merges --oneline
0d83a2ecd Roll src/third_party/skia 15dc59e65cfa..c5ee499f2c59 (2 commits) (flutter/engine#8298)
42d062f95 [fuchsia] Add missing trace macros (flutter/engine#8297)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

hyjfine committed:
Let CupertinoTextField's clear button also call onChanged (#29474)

Commit:
Lazy cache 4 (#29785)

zanderso committed:
[fuchsia] Fix flutter_tool BUILD.gn deps (#29942)

amirh committed:
Limit the semantic nodes ID range to 2^16 (#29928)

This PR limits the framework generated semantic node IDs to be smaller than 2^16, this allows to safely generate semantic node in the engine with IDs >= 2^16 avoiding ID collision (which is done in flutter/engine#8250).

Commit:
Revert "Lazy cache 4 (#29785)" (#29985)

Piinks committed:
Linking Higher & Lower Class Docs (#29758)

* Adding linking between higher and lower level classes in the API Docs. ref:#22859

* Fixed links between classes.

dnfield committed:
Avoid overwriting task result for non-leak checkers (#29989)

Commit:
ensure that fe server is always built with tool (#29768)

Commit:
Watch wildcard directories in addition to asset bundle (#29883)

rami-a committed:
Fix issue with account drawer header arrow rotating when setState is called (#29980)

The UserAccountsDrawerHeader had a bug with the arrow rotating whenever setState was called on its parent. This change makes sure that we check that the state actually changed before allowing the animation to trigger.

Fixes #25801

ds84182 committed:
Allow PointerEnterEvent and PointerExitEvents to be created from any PointerEvent (#28602)

Avoids an implicit downcast + crash in MouseTracker, since the last
pointer event can be a Move or Down event, in addition to Hover.

LongCatIsLooong committed:
Remove unwanted gap between navigation bar and safe area's child (#29943)

Remove the additional top padding from CupertinoPageScaffold's MediaQuery when the navigation bar is opaque (as the padding was already consumed by the navigation bar).

Related Issue: #29136

justinmc committed:
Docs explaining how to avoid the problem of height toggling with error (#29811)

Documentation that explains a common use case seen in Github issues where users were surprised by TextFormField growth to accommodate validation messages.

LongCatIsLooong committed:
update CupertinoSwitch documentation (#29987)

Documented CupertinoSwitch.value that it can't be null and added an assertion value != null in the constructor.

Related issue: #12948

liyuqian committed:
Change `-c` to `--enable-asserts` (#30055)

Unblock the engine roll that includes the Dart roll that removes `-c`.

xster committed:
Let CupertinoPageScaffold have tap status bar to scroll to top (#29946)

Commit:
Lazy cache 5: The Empire Strikes Back (#29986)

Commit:
Include platformViewId in semantics tree for iOS (#29304)

Include the platformViewId of PlatformViews in the semantics tree. The accessibility bridge in the engine can use this id to steal the semantics nodes from the actual platform view and stick them into Flutter's semantics tree.

It is the iOS PlatformView counter part of https://github.com/flutter/flutter/pull/28953. It reverts the change in https://github.com/flutter/flutter/pull/28953/commits/5b5d6e89ec50f8f51766fb778d0e244f183a5000 and https://github.com/flutter/flutter/pull/28953/commits/03fd797eb8955561c2654b51035175202a25da9d.

https://github.com/flutter/flutter/issues/29302

Commit:
ensure packages file is updated when using build_runner (#29885)

Piinks committed:
Embedding new diagrams for API Docs (#29697)

* Embedding new diarams for API docs: Container, ListView, Align.

* Moving the diagrams to be enclosed within the snippet windows.

* Fixed wording for the alt image text.

Commit:
Pass FLUTTER_TOOL_ARGS to snapshot command. (#29938)

engine-flutter-autoroll committed:
Roll engine 0d83a2ecd1bb..3a3f707d5a9b (25 commits) (#30063)


https://github.com/flutter/engine/compare/0d83a2ecd1bb...3a3f707d5a9b

git log 0d83a2ecd1bb78f8e1f0ed9719ff3751cdfdd064..3a3f707d5a9b17388b60ff10b9f65b0c6a3c4134 --no-merges --oneline
3a3f707d5 Reland &#34;Allow specification of std::functions as native entrypoints from Dart code.&#34; (flutter/engine#8329)
d4275d9ee Roll src/third_party/skia 576eb87a2d2d..99ccc0ca87e6 (5 commits) (flutter/engine#8328)
3a415c4f1 Map glfw into third_party, and roll buildroot (flutter/engine#8308)
ce9ea5869 [fuchsia] Disable FML_TRACE_COUNTER events to unblock roll (flutter/engine#8325)
48496588c Roll src/third_party/dart 8a92d2a8d9..991c9da720 (2 commits)
20c241ac3 Roll src/third_party/skia c476e5da4fef..576eb87a2d2d (7 commits) (flutter/engine#8324)
cadcf1c58 Roll src/third_party/dart f3fd1943fc..8a92d2a8d9 (16 commits)
66141eb8e Roll src/third_party/skia d1c271bd39f0..c476e5da4fef (2 commits) (flutter/engine#8322)
e5b31cdd9 Roll src/third_party/skia cec20280b3fb..d1c271bd39f0 (3 commits) (flutter/engine#8321)
1daff5ce1 Roll src/third_party/skia bf341ae88c83..cec20280b3fb (3 commits) (flutter/engine#8320)
87db585ad Roll src/third_party/skia 45d5f702133e..bf341ae88c83 (2 commits) (flutter/engine#8316)
fd7d7fa5a Add a11y support for embedded iOS platform view (flutter/engine#8156)
f64ee01e1 Roll src/third_party/skia d2ca31218bc4..45d5f702133e (11 commits) (flutter/engine#8315)
f521df383 Fixed service isolate not being initialized correctly due to bad name (flutter/engine#8251)
80b825ce8 Roll src/third_party/dart 093c2909fe..f3fd1943fc (13 commits) (flutter/engine#8310)
7e77d5c48 Revert &#34;Allow specification of std::functions as native entrypoints from Dart code. (#8309)&#34; (flutter/engine#8312)
400a86a62 Allow specification of std::functions as native entrypoints from Dart code. (flutter/engine#8309)
51f23fedf [vulkan] Add FUCHSIA external sem/mem extensions
78de8dcb4 Enable lambda like native callbacks in tests and add support for custom entrypoints. (flutter/engine#8299)
2812c7db4 Roll src/third_party/skia 62fd6c411622..d2ca31218bc4 (9 commits) (flutter/engine#8307)
95f9134e9 Roll src/third_party/skia d90004516a63..62fd6c411622 (4 commits) (flutter/engine#8306)
358273bd2 Roll src/third_party/skia 33211b2c7a02..d90004516a63 (1 commits) (flutter/engine#8305)
2804057bf Roll src/third_party/skia 80dd599e91d0..33211b2c7a02 (1 commits) (flutter/engine#8303)
a673bbf6f Roll src/third_party/skia c5ee499f2c59..80dd599e91d0 (5 commits) (flutter/engine#8301)
d88037dc0 Roll src/third_party/dart fa74184b7a..093c2909fe (71 commits)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

xster committed:
Cupertino localization step 1.5: fix a resource mismatch in cupertino_en.arb (#29821)

Commit:
Make timeout durations configurable (#30053)

engine-flutter-autoroll committed:
Roll engine 3a3f707d5a9b..76200562779e (2 commits) (#30070)


https://github.com/flutter/engine/compare/3a3f707d5a9b...76200562779e

git log 3a3f707d5a9b17388b60ff10b9f65b0c6a3c4134..76200562779e49d34e62056b66800ac870197092 --no-merges --oneline
762005627 Android Embedding PR22: Polish - FlutterActivity Intent factories, FlutterFragment control of render modes, FlutterSurfaceView transparent until rendering is ready. (flutter/engine#8317)
6bd697dd4 Fix &#34;PointerEvent&#34; flow end event (flutter/engine#8319)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

Commit:
Update to latest matcher (#30019)

Nizarius committed:
CupertinoTextField: added ability to change placeholder color (#28001)

shihaohong committed:
Added Hero zero-frame animation documentation (#30048)

engine-flutter-autoroll committed:
Roll engine 76200562779e..37947f961a62 (2 commits) (#30074)


https://github.com/flutter/engine/compare/76200562779e...37947f961a62

git log 76200562779e49d34e62056b66800ac870197092..37947f961a6242d2efc05a55a4319b120764a403 --no-merges --oneline
37947f961 Allow delegation of a11y events from nodes that were not yet traversed (flutter/engine#8333)
deeb7a747 Roll src/third_party/skia 99ccc0ca87e6..8d2c19554e4a (11 commits) (flutter/engine#8332)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

Commit:
Move binarySearch implementation in animated_list to foundation/collection.dart. (#29860)

engine-flutter-autoroll committed:
Roll engine 37947f961a62..c4d14a00716f (11 commits) (#30094)


https://github.com/flutter/engine/compare/37947f961a62...c4d14a00716f

git log 37947f961a6242d2efc05a55a4319b120764a403..c4d14a00716fa616b69d51bab957aee483bdc9c2 --no-merges --oneline
c4d14a007 Roll src/third_party/skia e4c67058ddb1..01a065884b7d (1 commits) (flutter/engine#8344)
5983b7adf Roll src/third_party/dart ffee99d79b..cf32584870 (6 commits)
84c62b4c7 Build Windows shell (flutter/engine#8331)
6415277d0 Roll src/third_party/skia 8d2c19554e4a..e4c67058ddb1 (1 commits) (flutter/engine#8341)
a0110107e Roll src/third_party/dart 991c9da720..ffee99d79b (7 commits)
2098398e1 Cleanups to run_tests.sh script (flutter/engine#8337)
5c99138fa Build GLFW from source for Linux shell (flutter/engine#8327)
68002458f Fix Windows build. (flutter/engine#8336)
6d8b83677 Remove use of epoxy from Linux shell (flutter/engine#8334)
4efc32163 Add super call in FLEView reshape (flutter/engine#8335)
cb8eb801a Allow embedders to specify their own task runner interfaces. (flutter/engine#8273)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

franciscojma86 committed:
Adds the keyboard mapping for Linux (#29993)

christopherfujino committed:
Move spinner `_defaultSlowWarning` message to a new line (#30071)

* write newline before adding slow restart message to spinner
* update existing test

Commit:
Add more test coverage to image handling (#30078)

Commit:
skip .dart_tool folders when running update-packages (#30082)

Hixie committed:
Update PULL_REQUEST_TEMPLATE.md (#29621)

This tries to make it more obvious when someone checks the "tests" box without actually having added tests.

Commit:
Fix a misuse of matchesGoldenFile() in the physical_model_test. (#30141)

Fixed a misuse of matchesGoldenFile() in the physical_model_test so that it correctly reports a failed test on Linux (and skips the other platforms).

shihaohong committed:
Add link to hero animation page (#30059)

Commit:
Intercept errors thrown by synchronous Completers in image resolution. (#30139)

liyuqian committed:
Error message for setting shaderWarmUp too late (#30145)

Developers may get confused by setting PaintingBinding.shaderWarmUp in
the wrong place. The added assert and error message help avoid that.

shihaohong committed:
Set Max Height for ListTile trailing and leading widgets (#29771)

dnfield committed:
roll engine to 82765aa77db9621dfbc50801ee2709aa0a00e04d (#30198)

Commit:
disable jit snapshot until we know what is going on here (#30204)

goderbauer committed:
Part 1: Improve Overlay API (#28747)

Commit:
Made the showMenu() position parameter required  (#30206)

Made the showMenu() position parameter required as it doesn't make sense to show a menu without indicating where it should be shown. Also added a test to verify this.

engine-flutter-autoroll committed:
Roll engine 82765aa77db9..a1a2129bc469 (13 commits) (#30209)


https://github.com/flutter/engine/compare/82765aa77db9...a1a2129bc469

git log 82765aa77db9621dfbc50801ee2709aa0a00e04d..a1a2129bc4696e66510be984fd7908eae51eb98f --no-merges --oneline
a1a2129bc GN Format all files in the engine. (flutter/engine#8369)
82c910414 Remove old Fuchsia external mem,sem extensions
6d1a6a4f4 Allow per-platform customization of the default Skia font manager (flutter/engine#8358)
3de50b8d2 Roll src/third_party/skia 84a53268af92..7eeb74fdfd84 (3 commits) (flutter/engine#8367)
65837ab32 Roll src/third_party/skia a2cec203eea8..84a53268af92 (1 commits) (flutter/engine#8366)
ada0a8e2b Roll src/third_party/skia 57b7a792bb6f..a2cec203eea8 (1 commits) (flutter/engine#8365)
9729f1ae3 Roll src/third_party/skia bf4229cc10a1..57b7a792bb6f (5 commits) (flutter/engine#8364)
23e0c3758 Roll src/third_party/skia 01fe26360660..bf4229cc10a1 (6 commits) (flutter/engine#8363)
fd0b44045 Roll src/third_party/skia cc8c7bee7925..01fe26360660 (10 commits) (flutter/engine#8361)
96c975163 Rename threshold to access_threshold (flutter/engine#8354)
09a67fe60 update buildroot dep for https://github.com/flutter/buildroot/commit/e2ca4571fa39be20ab1bf67c65d4700612756af2 (flutter/engine#8359)
602b61213 Reduce z-fighting on Scenic (flutter/engine#8318)
b53e436f9 Roll src/third_party/dart a977273262..f29e100f42 (8 commits)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

Commit:
add missing test case and handle wildcard removal (#30205)

dacaiguoguogmail committed:
Set custom flutter_assets by add FLTAssetsPath to AppFrameworkInfo.plist (#29499)

* Get FLTAssetsPath from AppFrameworkInfo.plist

xster committed:
Make sure everything in the Cupertino page transition can be linear when back swiping (#28629)

xster committed:
Cupertino localization step 3: in-place move some material tools around to make room for cupertino (#29644)

Commit:
make sure flutter test asks for cache upgrades (#30216)

engine-flutter-autoroll committed:
Roll engine a1a2129bc469..86f4b741b863 (5 commits) (#30223)


https://github.com/flutter/engine/compare/a1a2129bc469...86f4b741b863

git log a1a2129bc4696e66510be984fd7908eae51eb98f..86f4b741b863aef5586f03f4d250f1d789fde2be --no-merges --oneline
86f4b741b Move libdart selection into its own target in //flutter/runtime. (flutter/engine#8373)
f39cf3774 Add a GN format presubmit. (flutter/engine#8371)
d6e2d8527 Roll src/third_party/skia 7eeb74fdfd84..73c6c9d26870 (8 commits) (flutter/engine#8372)
8412c199a Enable Linux shell build (flutter/engine#8233)
133901199 libtxt: track the start and end x positions of glyph blobs for more accurate rendering of text decorations (flutter/engine#8368)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

Commit:
add coverage benchmark (#30222)

engine-flutter-autoroll committed:
Roll engine 86f4b741b863..ca7623eb39d7 (5 commits) (#30229)


https://github.com/flutter/engine/compare/86f4b741b863...ca7623eb39d7

git log 86f4b741b863aef5586f03f4d250f1d789fde2be..ca7623eb39d74a8cbdd095fcc7db398267b6928f --no-merges --oneline
ca7623eb3 Handle null values in TextInputConfiguration.actionLabel JSON (flutter/engine#8377)
d0239bd83 Roll src/third_party/skia 73c6c9d26870..59b733715141 (3 commits) (flutter/engine#8378)
71d59cde9 Check for hover motion events in AndroidTouchProcessor (flutter/engine#8376)
fdcbb40fe [flutter_tester] Accept --icu-data-file-path (flutter/engine#8374)
e633b26c5 Introduce unit tests and refactor web dart:ui into &#34;package&#34; (flutter/engine#8346)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

Commit:
Ensure that flutter run/drive/test/update_packages only downloads required artifacts (#30075)

Commit:
Revert "Ensure that flutter run/drive/test/update_packages only downloads required artifacts (#30075)" (#30232)

engine-flutter-autoroll committed:
Roll engine ca7623eb39d7..b75472c106d0 (5 commits) (#30231)


https://github.com/flutter/engine/compare/ca7623eb39d7...b75472c106d0

git log ca7623eb39d74a8cbdd095fcc7db398267b6928f..b75472c106d0cd0aab45de6bc16e0d6aa6d418cf --no-merges --oneline
b75472c10 Add missing import to functional for Windows. (flutter/engine#8382)
9d7167516 Ensure OpacityLayer to have a single child (flutter/engine#8362)
131cc625a Allow native entrypoint registration for runtime unittests. (flutter/engine#8379)
5e124ffc1 Fix typos (flutter/engine#8339)
972afdc92 Allow running runtime_unittests in AOT mode. (flutter/engine#8375)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
Roll engine b75472c106d0..8dabd60a30d0 (2 commits) (#30236)


https://github.com/flutter/engine/compare/b75472c106d0...8dabd60a30d0

git log b75472c106d0cd0aab45de6bc16e0d6aa6d418cf..8dabd60a30d048bf90b4bc0b29914ea00bda182c --no-merges --oneline
8dabd60a3 Remove unused DartVM::IsKernelMapping (flutter/engine#8381)
63b9d257c Roll src/third_party/dart f29e100f42..1e82544b07 (41 commits)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

shihaohong committed:
Added helpful Material assert message  (#30219)

Clarified an assert for when a Material widget's backgroundColor is null when its type isn't MaterialType.transparency

engine-flutter-autoroll committed:
Roll engine 8dabd60a30d0..28f8453629f9 (3 commits) (#30255)


https://github.com/flutter/engine/compare/8dabd60a30d0...28f8453629f9

git log 8dabd60a30d048bf90b4bc0b29914ea00bda182c..28f8453629f951d41437aa16bb09957643f1e850 --no-merges --oneline
28f845362 Roll src/third_party/skia c9822a1818ba..a6dd1ebd42cf (1 commits) (flutter/engine#8388)
2e1f491ba Roll src/third_party/skia 265fd3de3527..c9822a1818ba (1 commits) (flutter/engine#8386)
b466aba58 Roll src/third_party/skia 59b733715141..265fd3de3527 (1 commits) (flutter/engine#8384)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
ae024d755 Roll src/third_party/skia a6dd1ebd42cf..2fdbeae32785 (1 commits) (flutter/engine#8389) (#30257)


https://github.com/flutter/engine/compare/28f8453629f9...ae024d755518

git log 28f8453629f951d41437aa16bb09957643f1e850..ae024d755518019f3618931fbc47d3f27e84fad5 --no-merges --oneline
ae024d755 Roll src/third_party/skia a6dd1ebd42cf..2fdbeae32785 (1 commits) (flutter/engine#8389)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
be6313d87 Roll src/third_party/skia 2fdbeae32785..83edfbe736fd (1 commits) (flutter/engine#8390) (#30263)


https://github.com/flutter/engine/compare/ae024d755518...be6313d87545

git log ae024d755518019f3618931fbc47d3f27e84fad5..be6313d87545874a6108edb2a311f30d9a027472 --no-merges --oneline
be6313d87 Roll src/third_party/skia 2fdbeae32785..83edfbe736fd (1 commits) (flutter/engine#8390)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
f3ec76745 Roll src/third_party/skia 83edfbe736fd..acb4829c1be4 (1 commits) (flutter/engine#8391) (#30266)


https://github.com/flutter/engine/compare/be6313d87545...f3ec767458f1

git log be6313d87545874a6108edb2a311f30d9a027472..f3ec767458f12bb3099248fdc57d6c0d1051f042 --no-merges --oneline
f3ec76745 Roll src/third_party/skia 83edfbe736fd..acb4829c1be4 (1 commits) (flutter/engine#8391)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

zanderso committed:
[fuchsia_tester] Plumb through the location of icudtl (#30218)

engine-flutter-autoroll committed:
Roll engine f3ec767458f1..a850016454f6 (4 commits) (#30312)


https://github.com/flutter/engine/compare/f3ec767458f1...a850016454f6

git log f3ec767458f12bb3099248fdc57d6c0d1051f042..a850016454f6d33da64d03dbd1ea71fd0fac299f --no-merges --oneline
a85001645 Roll src/third_party/skia bb482ab871a2..73cc4e8d5fcf (1 commits) (flutter/engine#8395)
aba61e8fb Roll src/third_party/skia 5ee611b26f23..bb482ab871a2 (4 commits) (flutter/engine#8394)
82071e916 Don&#39;t access a11y APIs with reflection starting Android P. (flutter/engine#8393)
d9f16111a Roll src/third_party/skia acb4829c1be4..5ee611b26f23 (1 commits) (flutter/engine#8392)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

Commit:
Fix OutlineInputBorder crash (#30123)

dkwingsmt committed:
Simplify logic of TapGestureRecognizer (#30227)

Refactors the logic of TapGestureRecognizer, making the calling dependency unidirectional between resolve(accept) and checkUp.

engine-flutter-autoroll committed:
Roll engine a850016454f6..40eb74c50b06 (2 commits) (#30335)


https://github.com/flutter/engine/compare/a850016454f6...40eb74c50b06

git log a850016454f6d33da64d03dbd1ea71fd0fac299f..40eb74c50b0630d0a5fbf8586febe6581b893b66 --no-merges --oneline
40eb74c50 Make the resource context primary on iOS (flutter/engine#8387)
d8d9855df Roll src/third_party/skia 73cc4e8d5fcf..de206c75c2c8 (2 commits) (flutter/engine#8396)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
6c1b0b83c Roll src/third_party/dart 1e82544b07..75b2f9f919 (26 commits) (#30337)

LongCatIsLooong committed:
update sample code analyzer regexp & test case (#30201)

updates the regex the analyzer uses, so it should be able to recognize expressions such as
const Foo a = Foo(); as "other code" rather than a constructor call.

shihaohong committed:
Added assert to prevent complete ListTile trailing/leading horizontal expansion (#30212)

goderbauer committed:
Make heroes fly on pushReplacement (#30228)

engine-flutter-autoroll committed:
Roll engine 6c1b0b83c2b3..c991647404f8 (2 commits) (#30345)


https://github.com/flutter/engine/compare/6c1b0b83c2b3...c991647404f8

git log 6c1b0b83c2b3259cf7b23c2f6d6fc5cfdb924b61..c991647404f8117dab2ca4ed49aa3ba48a634240 --no-merges --oneline
c99164740 Separate the data required to bootstrap the VM into its own class. (flutter/engine#8397)
8c9aca440 Roll src/third_party/skia de206c75c2c8..b27667a15a15 (14 commits) (flutter/engine#8399)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
Roll engine c991647404f8..79a988f5da46 (2 commits) (#30350)


https://github.com/flutter/engine/compare/c991647404f8...79a988f5da46

git log c991647404f8117dab2ca4ed49aa3ba48a634240..79a988f5da4665980c6cc6916403e76b53f1e3e0 --no-merges --oneline
79a988f5d Roll src/third_party/skia b27667a15a15..3f6a2ac63670 (1 commits) (flutter/engine#8401)
cfe8675ff Delay platform view removal to submitFrame. (flutter/engine#8380)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
3796d9891 [ui] Add null check in FontWeight.lerp (flutter/engine#8274) (#30352)


https://github.com/flutter/engine/compare/79a988f5da46...3796d989183d

git log 79a988f5da4665980c6cc6916403e76b53f1e3e0..3796d989183d588b17a110d8aaf74906c4e1c8f0 --no-merges --oneline
3796d9891 [ui] Add null check in FontWeight.lerp (flutter/engine#8274)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

a14n committed:
no need .toList() before .join() (#30304)

engine-flutter-autoroll committed:
b572ade35 Roll src/third_party/skia 3f6a2ac63670..1acf79a3e5b8 (1 commits) (flutter/engine#8403) (#30355)


https://github.com/flutter/engine/compare/3796d989183d...b572ade3550b

git log 3796d989183d588b17a110d8aaf74906c4e1c8f0..b572ade3550b3f04ccb8c18d5bc4d5804f479c03 --no-merges --oneline
b572ade35 Roll src/third_party/skia 3f6a2ac63670..1acf79a3e5b8 (1 commits) (flutter/engine#8403)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (bmparr@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
6675f71df Roll src/third_party/skia 1acf79a3e5b8..d962070feb84 (4 commits) (flutter/engine#8404) (#30365)


https://github.com/flutter/engine/compare/b572ade3550b...6675f71dfb22

git log b572ade3550b3f04ccb8c18d5bc4d5804f479c03..6675f71dfb22a0b4fdab6332e37843f78c33c9e6 --no-merges --oneline
6675f71df Roll src/third_party/skia 1acf79a3e5b8..d962070feb84 (4 commits) (flutter/engine#8404)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

Piinks committed:
Updating sample code for raised button to match forthcoming diagram. (#30348)

legalcodes committed:
Fix minor typo (#30353)

engine-flutter-autoroll committed:
Roll engine 6675f71dfb22..f7b4903d7c99 (2 commits) (#30393)


https://github.com/flutter/engine/compare/6675f71dfb22...f7b4903d7c99

git log 6675f71dfb22a0b4fdab6332e37843f78c33c9e6..f7b4903d7c9913447af3aaebc4308d1a1e2a4dde --no-merges --oneline
f7b4903d7 Revert &#34;Separate the data required to bootstrap the VM into its own class. (#8397)&#34; (flutter/engine#8406)
f835789dd Roll src/third_party/skia d962070feb84..70d8f4f3acbb (1 commits) (flutter/engine#8405)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

xster committed:
Add needed singular resources for cupertino_en.arb (#30160)

amirh committed:
Make sure _handleAppFrame is only registered once per frame (#30346)

There were 2 possible scenarios in which _handleAppFrame is added more than once as a frame callback. When this happens it is possible that the second invocation will try to access _nextFrame.image when _nextFrame is null and crash. The 2 scenarios are:

Scenario 1

A GIF frame is decoded and a Flutter frame is executed before it's time to show the next GIF frame.
The timer that's waiting for enough time to elapse is invoked, and schedules a callback for the next Flutter frame(here).
Before the next Flutter frame is executed, MultiFrameImageStreamCompleter#removeListener is called followed by ``MultiFrameImageStreamCompleter#addListenerthat is invoking_decodeNextFrameAndSchedule` which is adding `_handleAppFrame` again as a next frame callback.
Scenario 2
removeListener and addListener are called multiple times in succession, every call to addListener can result in another registration of _handleAppFrame to the next Flutter frame callbacks list.

This patch fixes the issue by guarding against a second registration of _handleAppFrame.

engine-flutter-autoroll committed:
Roll engine f7b4903d7c99..4a8c7dceeb4b (5 commits) (#30399)


https://github.com/flutter/engine/compare/f7b4903d7c99...4a8c7dceeb4b

git log f7b4903d7c9913447af3aaebc4308d1a1e2a4dde..4a8c7dceeb4b7c710e0b22090cc3bc26f458a3a9 --no-merges --oneline
4a8c7dcee Roll src/third_party/skia f2d9b865d736..71c0e9422038 (13 commits) (flutter/engine#8413)
efbecd7e8 [txt] Add back FontCollection::SetDefaultFontManager (flutter/engine#8410)
5f7711449 Roll src/third_party/dart 75b2f9f919..14fd3b7d6a (23 commits)
f27c5730a [fuchsia] Exclude glfw from the Fuchsia host build (flutter/engine#8407)
853ac1bdf Roll src/third_party/skia 70d8f4f3acbb..f2d9b865d736 (11 commits) (flutter/engine#8408)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (chinmaygarde@google.com), and stop
the roller if necessary.

dnfield committed:
Draggable Scrollable sheet (#30058)

* Draggable Scrollable sheet

xster committed:
Cupertino localization step 4: let generated date localization combine material and cupertino locales (#29650)

engine-flutter-autoroll committed:
Roll engine 4a8c7dceeb4b..cc115500b9cb (6 commits) (#30421)


https://github.com/flutter/engine/compare/4a8c7dceeb4b...cc115500b9cb

git log 4a8c7dceeb4b7c710e0b22090cc3bc26f458a3a9..cc115500b9cb2b27d59627f202e6cd6bb87f49fc --no-merges --oneline
cc115500b Roll src/third_party/skia c9eeab3d85b9..e25b4472cdd9 (1 commits) (flutter/engine#8420)
20e45996d Roll src/third_party/skia 71c0e9422038..c9eeab3d85b9 (4 commits) (flutter/engine#8418)
a650790e4 Add scroll wheel support to desktop GLFW shell (flutter/engine#8416)
77d8e12e9 Roll src/third_party/dart 14fd3b7d6a..f6768b6fb3 (5 commits)
23fceaacd Added new Android embedding packages to javadoc generation. (flutter/engine#8411)
2f1786010 Pass environment defines to compile flutter platform step. (flutter/engine#8412)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (chinmaygarde@google.com), and stop
the roller if necessary.

mraleph committed:
Introduce --report-timings flag for flutter build aot command. (#30032)

This flag makes flutter build aot report timings for substeps (e.g.
frontend compilation and gen_snapshot) in a machine readable form.

goderbauer committed:
Bump dartdocs to 0.28.3 (#30451)

christopherfujino committed:
make shellcheck (linter) recommended changes to bin/flutter (#30456)

goderbauer committed:
Add 29 Widget of the Week videos (#30415)

goderbauer committed:
Moar Videos (#30452)

liyuqian committed:
Revert "Error message for setting shaderWarmUp too late (#30145)" (#30463)

This reverts commit ffbb335e35ae7416be98b72779baae6c9bd25b3b.

Reason for revert: flutter driver tests may have bugs in getting
first frame. Revert this until we figure out what happened.

HansMuller commented:
Thanks for the contribution, this looks like a nice addition to the Dismissable demo.

Per Flutter style we always [switch on enum values](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values). We also typically use the `dart:async' package and `await`. I'd suggest this for the `confirmDismiss` callback:
```
confirmDismiss: (DismissDirection dismissDirection) async {
  switch(dismissDirection) {
    case DismissDirection.endToStart:
      if (await _showConfirmationDialog(context, 'archive'))
        _handleArchive();
      break;

    case DismissDirection.startToEnd:
      if (await _showConfirmationDialog(context, 'delete'))
        _handleDelete();
      break;

    case DismissDirection.horizontal:
    case DismissDirection.vertical:
    case DismissDirection.up:
    case DismissDirection.down:
      assert(false);
  }
},
```

Also: would be nice if this feature could be disabled with this demo's Scaffold actions menu.

If you don't have time to continue working on this (completely understandable) please let us know and we'll help roll it across the finish line. Otherwise, it appears to be pretty close; just needs a few simple changes.

Thanks again for the contribution!

Commit:
Fix gallery API doc URL launcher (#30460)

zanderso committed:
[fuchsia] Fix isolate filter (#30458)

engine-flutter-autoroll committed:
309d078b5 Roll src/third_party/skia e25b4472cdd9..176da0bb8a10 (10 commits) (flutter/engine#8422) (#30441)


https://github.com/flutter/engine/compare/cc115500b9cb...309d078b597d

git log cc115500b9cb2b27d59627f202e6cd6bb87f49fc..309d078b597d3a38fe9bcef8d8bae7b56487e8eb --no-merges --oneline
309d078b5 Roll src/third_party/skia e25b4472cdd9..176da0bb8a10 (10 commits) (flutter/engine#8422)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (chinmaygarde@google.com), and stop
the roller if necessary.

goderbauer committed:
Be more explicit when ValueNotifier notifies (#30461)

goderbauer committed:
Random trivial fixes in the animation packages (#30276)

Piinks committed:
Updating sample code for BottomNavigationBar class for diagram. (#30453)

Piinks committed:
Embedding new raised button diagram. (#30398)

Commit:
Implement compute for async function (#16265) (#30275)

Taym95 commented:
Thank you for the response, yes I will make these changes.

Commit:
Use async await instead of promise and add possibility to disable confirmDismiss with scaffold actions menu

Commit:
Merge branch 'master' of github.com:Taym95/flutter
